pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -266,16 +267,23 @@ func (i *Instances) InstanceType(ctx context.Context, name types.NodeName) (stri
 		return "", err
 	}
 
-	return srvInstanceType(&srv.Server)
+	return srvInstanceType(i.compute, &srv.Server)
 }
 
-func srvInstanceType(srv *servers.Server) (string, error) {
+func srvInstanceType(client *gophercloud.ServiceClient, srv *servers.Server) (string, error) {
 	keys := []string{"name", "id", "original_name"}
 	for _, key := range keys {
 		val, found := srv.Flavor[key]
 		if found {
 			flavor, ok := val.(string)
 			if ok {
+				if key == "id" {
+					f, err := flavors.Get(client, flavor).Extract()
+					if err != nil {
+						return "", err
+					}
+					return f.Name, nil
```

Comment:
I don't understand why need force return f.Name, as we need some value to indicate it's the desired flavor..
then why we need force return name? 

https://github.com/openstack/nova/blob/master/nova/api/validation/parameter_types.py#L273
indicate flavor name min length is 1 ,so I think `name` will be satified and returned anyway

Reply:
The idea is to provide a more meaningful return, if we only have the ID to lookup. So when using `kubectl get nodes -L "node.kubernetes.io/instance-type"` we don't see the flavor id, but rather the name of it.

Reply:
Maybe @sbueringer could comment on this.

Reply:
ok, I think ID is also reasonable output ...
anyway, wait for other folks to comment

Reply:
@seanschneeweiss I guess "id" is available only till microversion 2.46 and "original_name" is the way forward in the latest version , so I'm ok to return name for consistency .

2 comments from my side
1. L282, we can just return the name if no error , else keep the existing behavior of returning 'id'
1. L274, even though , this is not of existing code change , could you check if "name" is valid key in the flavor object returned by server , I dont see it in compute API spec. If not, could you remove the same

Reply:
Agree with @ramineni on L282. For L274, please move `original_name` to the first and remove `name`

Reply:
Thank you. I've committed the proposed changes.